core/src/main/java/org/apache/carbondata/core/util/CarbonMetadataUtil.java

```diff
@@ -549,24 +562,23 @@ private static ValueEncoderMeta deserializeValueEncoderMeta(ByteBuffer byteBuffe
 
   }
 
-  private static WriterCompressModel getValueCompressionModel(ValueEncoderMeta[] encoderMetas) {
-    Object[] maxValue = new Object[encoderMetas.length];
-    Object[] minValue = new Object[encoderMetas.length];
-    int[] decimalLength = new int[encoderMetas.length];
-    Object[] uniqueValue = new Object[encoderMetas.length];
-    DataType[] aggType = new DataType[encoderMetas.length];
+  private static MeasurePageStatistics getMeasurePageStats(ValueEncoderMeta[] encoderMetas) {
+    Object[] max = new Object[encoderMetas.length];
+    Object[] min = new Object[encoderMetas.length];
+    int[] decimal = new int[encoderMetas.length];
+    Object[] nonExistValue = new Object[encoderMetas.length];
+    DataType[] types = new DataType[encoderMetas.length];
     byte[] dataTypeSelected = new byte[encoderMetas.length];
     for (int i = 0; i < encoderMetas.length; i++) {
-      maxValue[i] = encoderMetas[i].getMaxValue();
-      minValue[i] = encoderMetas[i].getMinValue();
-      decimalLength[i] = encoderMetas[i].getDecimal();
-      uniqueValue[i] = encoderMetas[i].getUniqueValue();
-      aggType[i] = encoderMetas[i].getType();
+      max[i] = encoderMetas[i].getMaxValue();
+      min[i] = encoderMetas[i].getMinValue();
+      decimal[i] = encoderMetas[i].getDecimal();
+      nonExistValue[i] = encoderMetas[i].getUniqueValue();
+      types[i] = encoderMetas[i].getType();
       dataTypeSelected[i] = encoderMetas[i].getDataTypeSelected();
     }
-    return ValueCompressionUtil
-        .getWriterCompressModel(maxValue, minValue, decimalLength, uniqueValue, aggType,
-            dataTypeSelected);
+
+    return MeasurePageStatistics.build(min, max, nonExistValue, decimal, types, dataTypeSelected);
```

Comment:
Better pass `ValueEncoderMeta` list to `build` method to avoid duplicate code in other places